File: TabExperiment/app/src/main/java/com/example/tabexperiment/PagerAdapter.java
package com.example.tabexperiment;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class PagerAdapter extends FragmentStateAdapter {
    private final int itemCount;

    public PagerAdapter(@NonNull FragmentManager fragmentManager,
                        @NonNull Lifecycle lifecycle,
                        int itemCount) {
        super(fragmentManager, lifecycle);
        this.itemCount = itemCount;
    }


    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return new TabFragment1();
            case 1:
                return new TabFragment2();
            case 2:
                return new TabFragment3();
        }
        return new TabFragment1();
    }

    @Override
    public int getItemCount() {
        return itemCount;
    }
}

File: OrderPlatform/app/src/main/java/com/example/orderplatform/adapter/PagerAdapter.kt
package com.example.orderplatform.adapter

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.orderplatform.fragment.InfoFragment
import com.example.orderplatform.fragment.OrderFragment
import com.example.orderplatform.fragment.ProductFragment

class PagerAdapter(
    fragmentManager: FragmentManager,
    lifecycle: Lifecycle,
    private val itemsCount: Int,
    private val context: Context
) :
    FragmentStateAdapter(fragmentManager, lifecycle) {

    override fun getItemCount(): Int {
        return itemsCount
    }

    override fun createFragment(position: Int): Fragment {
        when (position) {
            0, 1, 2, 3 -> {
                val bundle = Bundle()
                val productFragment = ProductFragment()
                bundle.putInt("kind", position)
                productFragment.arguments = bundle
                return productFragment
            }

            4 -> {
                return InfoFragment()
            }

            5 -> {
                return OrderFragment()
            }
        }
        return InfoFragment()
    }
}